src/tools/clang_tidy.jam

```diff
@@ -0,0 +1,81 @@
+#  Copyright (c) 2018 Arvid Norberg (arvid@libtorrent.org)
+#
+#  Use, modification and distribution is subject to the Boost Software
+#  License Version 1.0. (See accompanying file LICENSE_1_0.txt or
+#  http://www.boost.org/LICENSE_1_0.txt)
+
+import common ;
+import toolset ;
+import feature ;
+
+feature.extend toolset : clang_tidy ;
+
+generators.register-c-compiler clang_tidy.compile.c++ : CPP : OBJ : <toolset>clang_tidy ;
+generators.register-c-compiler clang_tidy.compile.c : C : OBJ : <toolset>clang_tidy ;
+generators.register-archiver clang_tidy.archive : OBJ : STATIC_LIB : <toolset>clang_tidy ;
+generators.register-linker clang_tidy.link : OBJ SEARCHED_LIB STATIC_LIB : EXE : <toolset>clang_tidy ;
+generators.register-linker clang_tidy.link.dll : OBJ SEARCHED_LIB STATIC_LIB : SHARED_LIB : <toolset>clang_tidy ;
+
+rule init ( version ? :  command * : options * ) {
+  command = [ common.get-invocation-command clang_tidy : clang-tidy
+    : $(command) ] ;
+
+  # Determine the version
+  if $(command) {
+    local command-string = \"$(command)\" ;
+    command-string = $(command-string:J=" ") ;
+    version ?= [ MATCH "version ([0-9.]+)"
+      : [ SHELL "$(command-string) --version" ] ] ;
+  }
+
+  local condition = [ common.check-init-parameters clang_tidy
+    : version $(version) ] ;
+
+  common.handle-options clang_tidy : $(condition) : $(command) : $(options) ;
+}
+
+###############################################################################
+# Flags
+
+toolset.flags clang_tidy.compile OPTIONS <cflags> ;
+toolset.flags clang_tidy.compile.c++ OPTIONS <cxxflags> ;
+
+toolset.flags clang_tidy.compile DEFINES <define> ;
+toolset.flags clang_tidy.compile INCLUDES <include> ;
+
+toolset.flags clang_tidy.compile OPTIONS <warnings>off : -w ;
+toolset.flags clang_tidy.compile OPTIONS <warnings>on  : -Wall ;
+toolset.flags clang_tidy.compile OPTIONS <warnings>all : -Wall -pedantic ;
+toolset.flags clang_tidy.compile OPTIONS <warnings-as-errors>on : -Werror ;
+
+toolset.flags clang_tidy.compile OPTIONS <rtti>off : -fno-rtti ;
+
+###############################################################################
+# C and C++ compilation
+
+TOUCH = [ common.file-creation-command ] ;
+
+actions compile.c++ {
+  "$(CONFIG_COMMAND)" -header-filter=*. -warnings-as-errors=* "$(>)" -- -x c++ $(OPTIONS) -D$(DEFINES) -I"$(INCLUDES)" \
```

Comment:
Does clang-tidy allow for response files as input?

If so, it would be nice to generate that here in the event that `DEFINES` or `INCLUDES` becomes large (to prevent blowing through max command line length):
e.g. `@@($(>:D).rsp:E=$(OPTIONS) -D$(DEFINES) -I"$(INCLUDES)")`

See [msvc.jam](https://github.com/boostorg/build/blob/develop/src/tools/msvc.jam#L605) for an example of the syntax and note that you may have to join each of the options with a newline depending on how clang's response files work.

Same for `compile.c` below.